app/src/test/java/com/kickstarter/viewmodels/BackingAddOnsFragmentViewModelTest.kt

```diff
@@ -522,6 +524,77 @@ class BackingAddOnsFragmentViewModelTest : KSRobolectricTestCase() {
                 }
     }
 
+    @Test
+    fun emptyState_whenNoAddOnsForShippingRule_shouldShowEmptyViewState() {
+        val shippingRuleAddOn = ShippingRuleFactory.germanyShippingRule()
+        val shippingRuleRw = ShippingRuleFactory.usShippingRule()
+        val addOn = RewardFactory.addOn().toBuilder()
+                .shippingRules(listOf(shippingRuleAddOn, shippingRuleAddOn, shippingRuleAddOn))
+                .shippingPreferenceType(Reward.ShippingPreference.RESTRICTED)
+                .build()
+        val listAddons = listOf(addOn, addOn, addOn)
+
+        val config = ConfigFactory.configForUSUser()
+        val currentConfig = MockCurrentConfig()
+        currentConfig.config(config)
+
+        setUpEnvironment(buildEnvironmentWith(listAddons, ShippingRulesEnvelope.builder().shippingRules(listOf(shippingRuleRw)).build(), currentConfig))
+
+        val rw = RewardFactory.rewardHasAddOns().toBuilder()
+                .shippingType(Reward.ShippingPreference.RESTRICTED.name.toLowerCase())
+                .shippingRules(listOf(shippingRuleRw))
+                .shippingPreferenceType(Reward.ShippingPreference.RESTRICTED)
+                .shippingPreference(Reward.ShippingPreference.RESTRICTED.name.toLowerCase())
+                .build()
+
+        val project = ProjectFactory.project().toBuilder().rewards(listOf(rw)).build()
+        val projectData = ProjectDataFactory.project(project, null, null)
+        val pledgeReason = PledgeFlowContext.forPledgeReason(PledgeReason.PLEDGE)
+
+        val bundle = Bundle()
+        bundle.putParcelable(ArgumentsKey.PLEDGE_PLEDGE_DATA, PledgeData.with(pledgeReason, projectData, rw))
+        bundle.putSerializable(ArgumentsKey.PLEDGE_PLEDGE_REASON, PledgeReason.PLEDGE)
+        this.vm.arguments(bundle)
+
+        this.addOnsList.assertValue(Triple(projectData, emptyList(), shippingRuleRw))
```

Comment:
something off with this test. If the empty state is emiting true, the addOnslist should be empty

Reply:
My bad, I did't realize the second value is the empty one! that's a great work :)